plugins/inputs/beat/beat.go

```diff
@@ -94,18 +94,49 @@ type Beat struct {
 	client *http.Client
 }
 
-func NewBeat() *Beat {
-	return &Beat{
-		URL:                  "http://127.0.0.1:5066",
-		CollectBeatStats:     true,
-		CollectLibbeatStats:  true,
-		CollectSystemStats:   true,
-		CollectFilebeatStats: true,
-		Method:               "GET",
-		Headers:              make(map[string]string),
-		HostHeader:           "",
-		Timeout:              internal.Duration{Duration: time.Second * 5},
+const (
+	defaultURL    = "http://127.0.0.1:5066"
+	defaultMethod = "GET"
+)
+
+var defaultCollectBeatStats = true
+var defaultCollectLibbeatStats = true
+var defaultCollectSystemStats = false
+var defaultCollectFilebeatStats = true
+var defaultTimeout = internal.Duration{Duration: time.Second * 5}
```

Comment:
Please don't declare them here. Either you declare those in `Init()` or you directly set the defaults there.

Reply:
Was necessary to make them vars in order to take the address of them. Thought it was cleaner to separate the default values from the code of setting the defaults.

Reply:
See my comment to change to a collection list in the form `collect = ["beat", "libbeat",...]`. You can use `internal/choice` to check the options an see what is in the user-setting.